### PR TITLE
bump tokenizers, fix added tokens fast

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Since tokenizer is being released, we can now fix the way tokens are added

### CODE REVIEW

File: src/transformers/convert_slow_tokenizer.py [hunk 1]
```diff
@@ -602,33 +602,12 @@ def tokenizer(self, proto):
...
-                if is_last_special:
-                    tokenizer.add_special_tokens(tokens)
-                else:
-                    tokenizer.add_tokens(tokens)
+        tokenizer.add_tokens(
+            [
+                AddedToken(token, normalized=False, special=special)
+                for id, token, special in sorted(spm_added_tokens, key=lambda x: x[0])
+            ]
+        )
```
Comment (LysandreJik) [inline src/transformers/convert_slow_tokenizer.py:610]: Much cleaner!

**Review by LysandreJik — APPROVED** [review]
Nice, the diff is nice to see!

**Comment by ArthurZucker** [comment]
Yeah still a few bugs to fix tho :( getting this ready for next release!

File: src/transformers/tokenization_utils_fast.py [hunk 2]
```diff
@@ -191,27 +184,7 @@ def __init__(self, *args, **kwargs):
             token for token in self.all_special_tokens_extended if token not in encoder and token not in tokens_to_add
```
Comment (itazap) [inline src/transformers/tokenization_utils_fast.py:184]:
```suggestion
tokens_to_add += [
          AddedToken(token, special=True) if isinstance(token, str) else token
          for token in self.all_special_tokens_extended
          if token not in encoder and token not in tokens_to_add
      ]```

File: src/transformers/tokenization_utils_fast.py [hunk 3]
```diff
@@ -191,27 +184,7 @@ def __init__(self, *args, **kwargs):
...
         ]
```
[the remark this answers is not in the document]
Comment (itazap) [reply src/transformers/tokenization_utils_fast.py:186]: ensures tokens are added as special if they are in all_special_tokens_extended (fixes test failure with `additional_special_tokens` param passed

[the remark this answers is not in the document]
Comment (itazap) [reply src/transformers/tokenization_utils_fast.py:186]: also somehow `all_special_tokens_extended` (using `special_tokens_map_extended`) has the `<mask>` token as `special=False` which causes a test failure

File: src/transformers/tokenization_utils_fast.py [hunk 4]
```diff
@@ -191,27 +184,7 @@ def __init__(self, *args, **kwargs):
...
 
-        if len(tokens_to_add) > 0:
-            # super hack: if a token.special is set, tokenizer ignores it for now so FIXME @ArthurZ
-            # Accumulate added tokens into batches of special/non-special tokens, because calling add_tokens() for
-            # individual tokens would repeatedly rebuild a trie, which can be slow.
-            is_last_special = None
-            tokens = []
-            special_tokens = self.all_special_tokens
```
Comment (itazap) [inline src/transformers/tokenization_utils_fast.py:200]: this used to ensure that tokens get added as special if they are in `all_special_tokens`. It even overwrote the default `special=False` param when creating an `AddedToken` such as the mask token in fast models (like here: https://github.com/huggingface/transformers/blob/2b789f27f383435b8db2fee3d10b0a1358c0c234/src/transformers/models/layoutxlm/tokenization_layoutxlm_fast.py#L233-L234 

But I'm not sure why only the mask token gets added like this in so many models and if we should just pass `special=True` there in each model?

Comment (ArthurZucker) [reply src/transformers/tokenization_utils_fast.py:200]: yeah we should make them special

File: src/transformers/tokenization_utils_fast.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply src/transformers/tokenization_utils_fast.py:186]: Mmmm but I'd rather we just take them from `added_tokens_decoder` raw, as it only stores added tokens!

File: src/transformers/tokenization_utils_fast.py (hunk 4 above)
Comment (ArthurZucker) [reply src/transformers/tokenization_utils_fast.py:200]: Actually you are right, I kep the logic to check special / change special if in all special tokens

Comment (ArthurZucker) [reply src/transformers/tokenization_utils_fast.py:205]: All tests pass thanks to you

**Pull request merged**